static/js/vue-cdr-access/tests/unit/facets.spec.js

```diff
@@ -115,94 +129,107 @@ describe('facets.vue', () => {
     });
 
     it("clears all selected facets if 'Clear Filters' button is clicked", async () => {
+        await router.push('/search/?collection=d77fd8c9-744b-42ab-8e20-5ad9bdf8194e');
         selected_facet.trigger('click');
-        await wrapper.vm.$nextTick();
-        expect(wrapper.vm.selected_facets).toEqual(['format=image']);
+        await flushPromises();
+        expect(wrapper.vm.selected_facets).toContain('format=image');
 
         wrapper.find('#clear-all').trigger('click');
-        await wrapper.vm.$nextTick();
+        await flushPromises();
         expect(wrapper.vm.selected_facets).toEqual([]);
     });
 
     it("clears a selected facet if it is unchecked", async () => {
+        await router.push('/search/?collection=d77fd8c9-744b-42ab-8e20-5ad9bdf8194e');
```

Comment:
how were these tests previous getting to the right page?

Reply:
Perhaps part of it was there were routes in the history from previous tests, not good. Now the router gets reset after every test. Not sure if this is part of it, but the routing is now all asynchronous, which it wasn't before.